deploy/docs/v2_migration_doc.md

```diff
@@ -136,7 +163,83 @@ kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheu
 kubectl apply -f https://raw.githubusercontent.com/prometheus-operator/prometheus-operator/release-0.43/example/prometheus-operator-crd/monitoring.coreos.com_thanosrulers.yaml
 ```
 
-### 3. Run upgrade script
+### 3. Prepare fluent-bit instance
+
+As `spec.selector` in fluent-bit Helm chart was modified it is required to recreate manually or delete existing DaemonSet
+with old version of `spec.selector` before upgrade.
+
+One of two strategy can be selected:
+
+- #### Recreating fluent-bit DaemonSet
+
+  Recreating fluent-bit DaemonSet with new `spec.selector` which may cause that applications logs and fluent-bit metrics
+  will not be available in the time of recreation. It usually takes several seconds.
+
+  To recreate a fluent-bit DaemonSet with a new `spec.selector`:
+
+  ```bash
+  kubectl get daemonset -l "app=fluent-bit,release=<RELEASE-NAME>" -o yaml | \
+  yq w - "items[*].spec.selector.matchLabels[app.kubernetes.io/name]" "fluent-bit" | \
+  yq w - "items[*].spec.selector.matchLabels[app.kubernetes.io/instance]" "<RELEASE-NAME>" | \
+  yq w - "items[*].spec.template.metadata.labels[app.kubernetes.io/name]" "fluent-bit" | \
+  yq w - "items[*].spec.template.metadata.labels[app.kubernetes.io/instance]" "<RELEASE-NAME>" | \
+  yq d - "items[*].spec.selector.matchLabels[app]" | \
+  yq d - "items[*].spec.selector.matchLabels[release]" | \
+  kubectl apply --force -f -
+  ```
+
+- #### Preparing temporary instance of fluent-bit
+
+  Create temporary instance of fluent-bit and delete DaemonSet with old version of `spec.selector`
+  which will cause that logs and metrics will be duplicated until temporary instance of fluent-bit will not be deleted
+  after upgrade.
```

Comment:
```suggestion
  Create temporary instance of fluent-bit and delete DaemonSet with old version of `spec.selector`.
  This will cause logs and metrics to be duplicated until temporary instance of fluent-bit is deleted
  after the upgrade is complete.
```